xbmc/input/linux/LIRC.cpp

```diff
@@ -268,7 +296,10 @@ void CRemoteControl::Update()
   }
 
   if (feof(m_file) != 0)
+  {
+    lock.unlock(); //Disconnect takes the lock
```

Comment:
I would use CSingleExit. You never know if anybody will add some code to the end of this function in the future.